main.ts

```diff
@@ -126,43 +129,99 @@ export class ThingsView extends ItemView {
 	async handleCheckboxClick(event: MouseEvent) {
 		const clickedCheckbox = event.target as HTMLInputElement;
 
-		const todoId = clickedCheckbox.attributes.getNamedItem("tid")?.value || ""
-		await this.completeTodoByJXA(todoId)
+		const todoId =
+			clickedCheckbox.attributes.getNamedItem("tid")?.value || "";
+		await this.completeTodoByJXA(todoId);
 
-		clickedCheckbox.parentNode?.detach()
+		clickedCheckbox.parentNode?.detach();
 
 		// things3 is too slow to refresh this immediately
-		this.refreshTodayView(3000)
+		this.refreshTodayView(3000);
 	}
 
 	refreshTodayView(delay?: number, notice = false) {
-		clearTimeout(this.refreshTimer)
+		clearTimeout(this.refreshTimer);
 
 		this.refreshTimer = setTimeout(() => {
 			this.getAndShowTodayTodos();
 			if (notice) {
-				new Notice("Today Refreshed")
+				new Notice("Today Refreshed");
 			}
 		}, delay);
 	}
 
 	getTodayListByJXA(): Promise<string> {
-		const getTodayListSct = `"function getTodayList() { let content = ''; Application('Things').lists.byId('TMTodayListSource').toDos().forEach(t => { let checked = t.status()=='open' ? '' : 'checked'; content += '<ul><input '+ checked +'  type="checkbox" class="things-today-checkbox" tid=\\"' + t.id() + '\\"><div style="display:contents"><a href=\\"things:///show?id=' + t.id() + '\\">' + t.name() + '</a></div></ul>'; }); return content; }; getTodayList();"`
+		const getTodayListScript = `
+			function getTodayList() {
+				let content = '';
+				const todos = Application('Things').lists.byId('TMTodayListSource').toDos();
+				const grouped = {};
+
+				todos.forEach(t => {
+					const area = t.area() ? t.area().name() : undefined;
+					let projectName = t.project() ? t.project().name() : undefined;
+					
+					if (!projectName) {
+						projectName = area ? area : 'No Project';
+					}
+
+					if (!grouped[projectName]) {
+						grouped[projectName] = [];
+					}
+					grouped[projectName].push(t);
+				});
+
+				for (const project in grouped) {
+					content += '<p style="font-weight:bold;border-bottom: 1px solid #333;padding:0;margin:0;margin-bottom:0.5rem;padding-bottom:0.25rem">' + project + '</p>';
+					content += '<ul style="display:grid;gap:0.5rem;list-style:none;padding:0;margin:0;padding-bottom:2rem" tid="' + grouped[project][0].id() + '">';
```

Comment:
i think these inline styles can go to the CSS file, im pretty new to obsiduan, im moving from notion as its slow and annoying! ;)